packages/ember/tests/routing/substates_test.js

```diff
@@ -1,1042 +1,880 @@
+  [`@test Don't enter loading route unless either route or template defined`](assert) {
+    let deferred = RSVP.defer();
 
-  let appController = container.lookup('controller:application');
-  equal(appController.get('currentPath'), 'grandma.loading', 'currentPath reflects loading state');
+    this.router.map(function() {
+      this.route('dummy');
+    })
+    this.add('route:dummy', Route.extend({
+      model() {
+        return deferred.promise;
+      }
+    }));
+    this.addTemplate('dummy', 'DUMMY');
 
-  run(momDeferred, 'resolve', {});
-  equal(jQuery('#app', '#qunit-fixture').text(), 'GRANDMA MOM');
-  equal(appController.get('currentPath'), 'grandma.mom', 'currentPath reflects final state');
-});
+    return this.visit('/').then(() => {
+      this.visit('/dummy');
+      let appController = this.getController('application');
 
-QUnit.test('Slow promises returned from ApplicationRoute#model don\'t enter LoadingRoute', function() {
-  expect(2);
+      assert.ok(appController.get('currentPath') !== 'loading', `
+        loading state not entered
+      `);
 
-  let appDeferred = RSVP.defer();
+      this.runTask(() => deferred.resolve());
+      let text = this.$('#app').text();
 
-  App.ApplicationRoute = Route.extend({
-    model() {
-      return appDeferred.promise;
-    }
-  });
+      assert.equal(text, 'DUMMY', `dummy template has been rendered`);
+    });
+  }
 
-  App.LoadingRoute = Route.extend({
-    setupController() {
-      ok(false, 'shouldn\'t get here');
-    }
-  });
+  ['@test Enter loading route only if loadingRoute is defined'](assert) {
+    let deferred = RSVP.defer();
 
-  bootApplication();
+    this.router.map(function() {
+      this.route('dummy');
+    });
 
-  equal(jQuery('#app', '#qunit-fixture').text(), '', 'nothing has been rendered yet');
+    this.add('route:dummy', Route.extend({
+      model() {
+        return deferred.promise;
+      }
+    }));
+    this.add('route:loading', Route.extend());
+    this.addTemplate('dummy', 'DUMMY');
 
-  run(appDeferred, 'resolve', {});
-  equal(jQuery('#app', '#qunit-fixture').text(), 'INDEX');
-});
+    return this.visit('/').then(() => {
+      this.visit('/dummy');
 
-QUnit.test('Don\'t enter loading route unless either route or template defined', function() {
-  delete templates.loading;
+      let appController = this.getController('application');
 
-  expect(2);
+      assert.equal(appController.get('currentPath'), 'loading',
+        `loading state entered`
+      );
 
-  let indexDeferred = RSVP.defer();
+      this.runTask(() => deferred.resolve());
+      let text = this.$('#app').text();
 
-  App.ApplicationController = Controller.extend();
+      assert.equal(text, 'DUMMY', `dummy template has been rendered`);
+    });
+  }
 
-  App.IndexRoute = Route.extend({
-    model() {
-      return indexDeferred.promise;
-    }
-  });
+  ['@test Slow promises returned from ApplicationRoute#model enter ApplicationLoadingRoute if present - NEW'](assert) {
```

Comment:
Shoot. The `- NEW` was when I was migrating tests over I could tell which of the two copies were failing. Will clean it out.